lib/fauna/query.rb

```diff
@@ -50,9 +50,24 @@ def event(resource, ts, action)
     ##
     # A let expression
     #
+    # Example: <code>Fauna.query { let(x: 2).in(add(1, var(:x))) }</code>.
+    #
     # Reference: {FaunaDB Basic Forms}[https://faunadb.com/documentation/queries#basic_forms]
-    def let(vars, in_expr)
-      { let: vars, in: in_expr }
+    def let(vars, in_expr = nil, &blk)
+      in_ = if blk.nil?
+        in_expr
+      else
+        dsl = DSLContext.new
+        dslcls = (class << dsl; include Query; self; end)
+
+        vars.keys.each do |v|
+          dslcls.send(:define_method, v) { var(v) }
+        end
+
+        DSLContext.eval_dsl(dsl, &blk)
```

Comment:
How does the usage of `dslcls` work? I see it used to define methods above to produce associated vars above, but then `eval_dsl` here uses the original `dsl`.